src/pinns_pde_solve.jl

```diff
@@ -293,35 +293,69 @@ function generator_training_sets(domains, discretization, bound_args, dict_indva
     [train_domain_set,train_bound_set]
 end
 
-# Convert a PDE problem into an PINNs problem
+function get_loss_function(pde_func, bc_funcs, train_sets)
+    # the points in the domain and on the boundary
+    train_domain_set, train_bound_set = train_sets
+
+    # coefficients for loss function
+    τb = length(train_bound_set)
+    τf = length(train_domain_set)
+
+    # Loss function for pde equation
+    function inner_domain_loss(phi, x, θ, derivative)
+        sum(pde_func(phi, x, θ, derivative))
+    end
+
+    # Loss function for initial and boundary conditions
+    function inner_bound_loss(f,phi,x,θ,derivative)
+        sum(f(phi, x, θ, derivative))
+    end
+
+    function loss_domain(θ, phi, derivative)
+        sum(abs2,inner_domain_loss(phi, x, θ, derivative) for x in train_domain_set)
+    end
+
+    function loss_boundary(θ, phi, derivative)
+       sum(sum(abs2,inner_bound_loss(f,phi,x,θ,derivative) for x in set) for (f,set) in zip(bc_funcs,train_bound_set))
+    end
+
+    # General loss function
+    function loss(θ, phi, derivative)
+        1.0f0/τf * loss_domain(θ, phi, derivative) + 1.0f0/τb * loss_boundary(θ, phi, derivative) #+ 1.0f0/τc * custom_loss(θ)
+    end
+    return loss
+end
+
+# Convert a PDE problem into OptimizationProblem
 function DiffEqBase.discretize(pde_system::PDESystem, discretization::PhysicsInformedNN)
     eq = pde_system.eq
     bcs = pde_system.bcs
     domains = pde_system.domain
     indvars = pde_system.indvars
     depvars = pde_system.depvars
-    dx = discretization.dx
-    dim = length(domains)
-
     # dictionaries: variable -> unique number
     dict_indvars = get_dict_vars(indvars)
     dict_depvars = get_dict_vars(depvars)
 
+    dim = length(domains)
+
     # extract pde
-    _pde_func = eval(extract_pde(eq,indvars,depvars, dict_indvars,dict_depvars))
+    pde_func = eval(extract_pde(eq,indvars,depvars, dict_indvars,dict_depvars))
     # extract initial and boundary conditions
     extract_bc_funcs, bound_args = extract_bc(bcs,indvars,depvars,dict_indvars,dict_depvars)
-
-    _bc_funcs = eval.(extract_bc_funcs)
+    bc_funcs = eval.(extract_bc_funcs)
 
     # generate training sets
     train_sets = generator_training_sets(domains, discretization, bound_args, dict_indvars)
 
-    return NNPDEProblem(_pde_func,_bc_funcs, train_sets, dim)
+    # get loss_function
+    loss_function = get_loss_function(pde_func,bc_funcs,train_sets)
+
+	return GalacticOptim.OptimizationProblem(loss_function, zeros(dim), p=nothing)
```

Comment:
Not an issue but not passing in `p` as `nothing` should work here

Reply:
yes, thanks